### ISSUE TITLE
Desktop capture

### ISSUE THREAD

**Issue by Erik-Bjerke** [wailsapp/wails#2955]
### Is your feature request related to a problem? Please describe.

In both Electron and Tauri I can easily capture the desktop. Currently if I try to screen capture from JavaScript it will fail because "the user doesn't give permission"

### Describe the solution you'd like

 I'd like to have Desktop Capture in Wails or a way to to handle the Webview to get permissions from the user.

### Describe alternatives you've considered

I've considered just using Electron or Tauri but Electron is a pain and Tauri is not of my liking.

### Additional context

https://www.electronjs.org/docs/latest/api/desktop-capturer

**Comment by leaanthony** [comment]
Hi there 👋 Thanks for opening this. If Tauri does it then the APIs should be available and we would be open to it for sure. Is this something you would like to implement?

**Comment by TheProcedural** [comment]
I've been working on getting it done through the API but I decided to give up (temporarily) due to some limitations. In the meanwhile I decided to do something a little more hacky and use Gstreamer, then pass the video stream using sockets.

I will revisit this in a near future though, if I have time I'll implement a proper Desktop Capturer.

Correcting what I initially said, Tauri is actually in the same state. But this particular "issue" is "easier" to handle with Wry than it is with go_WebView2

**Comment by leaanthony** [comment]
Glad you're making progress.
What makes it easier with wry?

**Comment by TheProcedural** [comment]
It's actually the same, it just was initially easier to find documentation, hence why I said "easier". After reading for a bit I can see why you said " If Tauri does it then the APIs should be available". I was actually not expecting this to be so easy in both haha.

I'm actually trying again with go_webview2 and it seems to be working so far. I'm still swamped with my job to work a on proper implementation for Wails right now. I'll try as soon as I can though!

**Comment by leaanthony** [comment]
Awesome. No rush. Let us know if you need a hand 👍

**Comment by TheProcedural** [comment]
I'm actually having an issue already, can I not work from Linux or am I' being dumb and missing something here? I searched the Wails docs on contributions but I can't find any instructions on requirements or setup.

![image](https://github.com/wailsapp/wails/assets/54407345/44898df6-a42e-4078-bf7c-3afbc7ae30f4)

**Comment by leaanthony** [comment]
Go-webview2 is a windows package

**Comment by TheProcedural** [comment]
There certainly is some confusion. Can I be pointed to some build instructions? Maybe I'm just being dumb here.

**Comment by TheProcedural** [comment]
Nevermind, I figured it out, the compile problems were related to vscodium somehow, which is extremely weird vscodium is doing something to my go installation when I open a project nested within another folder in this case "wails/v2" when I opened "v2" directly the problems went away. First time ever something so weird happens I need to find out why.

In any case, I made a wrong assumption when I thought the problems were connected, clearly they were not. Mad bad!
